### PR TITLE
docs: add documentation for API guardian

### PR DESCRIPTION
**Pull request by alan-agius4**
Relates to and blocked until: https://github.com/angular/angular-cli/pull/12010 is merged

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -274,3 +275,38 @@ changes to be accepted, the CLA must be signed. It's a quick process, we promise
...
+Example:
+```javascript 
+import * as foo from './foo';
+
+export { foo };
+```
+There are currently not supported by the API guardian. 
+To overcome this limitiation we created `_golden-api.ts` in certain packages.
+
+When adding a new API, it might be the case that you need to add it to the later mentioned file.
```
Comment (filipesilva) [inline CONTRIBUTING.md]: `later mentioned file` -> `to _golden-api.ts`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -274,3 +275,38 @@ changes to be accepted, the CLA must be signed. It's a quick process, we promise
 [individual-cla]: http://code.google.com/legal/individual-cla-v1.0.html
 [js-style-guide]: https://google.github.io/styleguide/jsguide.html
 [stackoverflow]: http://stackoverflow.com/questions/tagged/angular-devkit
+
+## <a name="public-api"></a> Updating the Public API
+Our Public API is protected with TS API Guardian. This is a tool that keeps track of public API surface of our packages.
+
+To test if your change effect the public API you need to run the API guardian on that particual package.
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 1 above)
Comment (alan-agius4) [reply CONTRIBUTING.md]: done

**Comment by alan-agius4** [comment]
Caretaker: CI is flaky.

**Pull request merged**